app/logic/volunteers.py

```diff
@@ -81,3 +81,13 @@ def addVolunteerToEventRsvp(user, volunteerEventID):
 
     except Exception as e:
         return False
+
+def setUserBackgroundCheck(user,bgType, checkPassed):
+    user = User.get_by_id(user)
+    requiredQuery = BackgroundCheck.select().where(BackgroundCheck.user == user, BackgroundCheck.type == bgType)
+
+    if not requiredQuery.exists():
+        BackgroundCheck.create(user = user, type = bgType, passBackgroundCheck = checkPassed) # Check if this is needed.
+    else:
+        update = BackgroundCheck.update(passBackgroundCheck = checkPassed).where(BackgroundCheck.user == user, BackgroundCheck.type == bgType)
+        update.execute()
```

Comment:
A shorter way with no conditionals since we don't need to maintain any history is to delete any entry with that type, and then insert a new one. Or you could use get_or_create instead of the initial select and edit the field and save.

Reply:
fixed

Reply:
Worked on this